junit/src/main/java/cucumber/runtime/junit/ExecutionUnitRunner.java

```diff
@@ -1,48 +1,43 @@
 package cucumber.runtime.junit;
 
-import cucumber.runtime.Runtime;
-import cucumber.runtime.model.CucumberScenario;
-import gherkin.formatter.model.Step;
+import cucumber.runner.Runner;
+import gherkin.events.PickleEvent;
+import gherkin.pickles.PickleStep;
 import org.junit.runner.Description;
 import org.junit.runner.notification.RunNotifier;
 import org.junit.runners.ParentRunner;
 import org.junit.runners.model.InitializationError;
 
-import java.util.ArrayList;
+import java.io.Serializable;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 /**
  * Runs a scenario, or a "synthetic" scenario derived from an Examples row.
  */
-public class ExecutionUnitRunner extends ParentRunner<Step> {
-    private final Runtime runtime;
-    private final CucumberScenario cucumberScenario;
+public class ExecutionUnitRunner extends ParentRunner<PickleStep> {
+    private final Runner runner;
+    private final PickleEvent pickleEvent;
     private final JUnitReporter jUnitReporter;
+    private final Map<PickleStep, Description> stepDescriptions = new HashMap<PickleStep, Description>();
     private Description description;
-    private final Map<Step, Description> stepDescriptions = new HashMap<Step, Description>();
-    private final List<Step> runnerSteps = new ArrayList<Step>();
 
-    public ExecutionUnitRunner(Runtime runtime, CucumberScenario cucumberScenario, JUnitReporter jUnitReporter) throws InitializationError {
+    public ExecutionUnitRunner(Runner runner, PickleEvent pickleEvent, JUnitReporter jUnitReporter) throws InitializationError {
         super(ExecutionUnitRunner.class);
```

Comment:
As far as I can tell this should be super(null). There is no reason to scan this class for annotations.